code/modules/mob/living/carbon/xenomorph/castes/facehugger/abilities_facehugger.dm

```diff
@@ -0,0 +1,133 @@
+// ***************************************
+// *********** Hug
+// ***************************************
+/datum/action/xeno_action/activable/pounce_hugger
+	name = "Pounce"
+	action_icon_state = "pounce"
+	desc = "Leap at your target and knock them down, if you jump close you will hug the target."
+	ability_name = "pounce"
+	cooldown_timer = 5 SECONDS
+	plasma_cost = 25
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_POUNCE,
+	)
+	use_state_flags = XACT_USE_BUCKLED
+	///How far can we leap.
+	var/range = 6
+	///Range for Hug
+	var/hug_range = 1
+	///For how long will we stun the victim
+	var/victim_paralyze_time = 1 SECONDS
+	///For how long will we freeze upon hitting our target
+	var/freeze_on_hit_time = 1 SECONDS
+	///How long is the windup before leap
+	var/windup_time = 1 SECONDS
+	///Where do we start the leap from
+	var/start_point
+
+
+/datum/action/xeno_action/activable/pounce_hugger/proc/pounce_complete()
+	SIGNAL_HANDLER
+	var/mob/living/carbon/xenomorph/caster = owner
+	caster.icon_state = "[caster.xeno_caste.caste_name] Walking"
+	UnregisterSignal(owner, list(COMSIG_XENO_OBJ_THROW_HIT, COMSIG_MOVABLE_POST_THROW, COMSIG_XENO_LIVING_THROW_HIT))
+
+
+/datum/action/xeno_action/activable/pounce_hugger/proc/obj_hit(datum/source, obj/target, speed)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/xenomorph/caster = owner
+	if(!istype(target, /obj/structure/table) && !istype(target, /obj/structure/rack))
+		target.hitby(caster, speed) //This resets throwing.
+	pounce_complete()
+
+/datum/action/xeno_action/activable/pounce_hugger/proc/mob_hit(datum/source, mob/living/M, test1)
```

Comment:
This abilities code is horrid, snowflake checks, no early returns, no documentation and what is there is bloat documentation, also looks like copypaste from runner pounce partially